checkbox-ng/checkbox_ng/launcher/subcommands.py

```diff
@@ -1267,6 +1270,91 @@ def __missing__(self, key):
+        parser.description = (
+            "Expand a given test plan: display all the jobs and templates "
+            "that are defined in this test plan and that would be executed "
+            "if ran. This is useful to visualize the full list of jobs and "
+            "templates for complex test plans that consist of many nested "
+            "parts with different 'include' and 'exclude' sections.\n\n"
+            "NOTE: the elements listed here are not sorted by execution "
+            "order. To see the execution order, please use the "
+            "'list-bootstrapped' command instead."
+        )
+        parser.add_argument("TEST_PLAN", help=_("test-plan id to expand"))
+        parser.add_argument(
+            "-f",
+            "--format",
+            type=str,
+            default="text",
+            help=_("output format: 'text' or 'json' (default: %(default)s)"),
+        )
+
+    def invoked(self, ctx):
+        self.ctx = ctx
+        session_title = "checkbox-expand-{}".format(ctx.args.TEST_PLAN)
+        self.sa.start_new_session(session_title)
+        tps = self.sa.get_test_plans()
+        if ctx.args.TEST_PLAN not in tps:
+            raise SystemExit("Test plan not found")
+        self.sa.select_test_plan(ctx.args.TEST_PLAN)
+        all_jobs_and_templates = [
+            unit
+            for unit in self.sa._context.state.unit_list
+            if unit.unit in ["job", "template"]
+        ]
+        tp = self.sa._context._test_plan_list[0]
+        tp_us = TestPlanUnitSupport(tp)
+        self.override_list = tp_us.override_list
+        jobs_and_templates_list = select_units(
+            all_jobs_and_templates,
+            [tp.get_mandatory_qualifier()] + [tp.get_qualifier()],
+        )
+
+        list_obj = []
+        for unit in jobs_and_templates_list:
+            obj = unit._raw_data.copy()
+            obj["unit"] = unit.unit
+            obj["id"] = unit.id  # To get the fully qualified id
+            obj[
+                "certification-status"
+            ] = self.get_effective_certification_status(unit)
+            if unit.template_id:
+                obj["template-id"] = unit.template_id
+            list_obj.append(obj)
+        if ctx.args.format == "json":
+            print(json.dumps(list_obj))
+        else:
+            for obj in list_obj:
+                if obj["unit"] == "template":
+                    print("Template '{}'".format(obj["template-id"]))
+                else:
+                    print("Job '{}'".format(obj["id"]))
+
+    def get_effective_certification_status(self, unit):
+        if unit.unit == "template":
+            unit_id = unit.template_id
+        else:
+            unit_id = unit.id
+        for regex, override_field_list in self.override_list:
+            if re.match(regex, unit_id):
+                for field, value in override_field_list:
+                    if field == "certification_status":
+                        return value
```

Comment:
Minor: see if you like something like this (also try it out, I'm writing this on github)
```suggestion
        with suppress(StopIteration):
            # search if the current unit has overridden fields via inclusion
            override_field_list = next(
                override_field_list
                for (regex, override_field_list) in self.override_list
                if re.match(regex, unit_id)
            )
            # search if among the overridden properties there is also the
            # certification status
            cert_status = next(
                value
                for (field, value) in override_field_list
                if field == "certification_status"
            )
            return cert_status
```

Reply:
the "closest" is right, which means inline override. (in the case of `certification_status`, it can be defined in the unit definition directly, in [a `certification_status_override`](https://github.com/canonical/checkbox/blob/main/checkbox-ng/plainbox/impl/unit/testplan.py#L791-L818) – which I realize now is not even documented, shame on us – or in an inline override)

Reply:
I'll skip this suggestion. I think it's harder to read than my dumb implementation (and it doubles the lines of code).